server/routerlicious/packages/lambdas/src/scribe/lambdaFactory.ts

```diff
@@ -175,34 +177,56 @@ export class ScribeLambdaFactory extends EventEmitter implements IPartitionLambd
 				lastCheckpoint.logOffset = -1;
 				const checkpointMessage = `Restoring checkpoint from latest summary. Seq number: ${lastCheckpoint.sequenceNumber}`;
 				context.log?.info(checkpointMessage, { messageMetaData });
-				Lumberjack.info(checkpointMessage, getLumberBaseProperties(documentId, tenantId));
+				Lumberjack.info(checkpointMessage, lumberProperties);
 			}
 		} else {
-			lastCheckpoint = JSON.parse(document.scribe);
+			let checkpoint;
+			let isLocalCheckpoint = false;
+
+			if (
+				this.localCheckpointEnabled &&
+				(this.localCheckpointCollection !== undefined ||
+					this.localCheckpointCollection !== null)
+			) {
+				// Search local database for checkpoint
+				Lumberjack.info(`Checking local DB for checkpoint.`, lumberProperties);
+				checkpoint = await this.localCheckpointCollection
+					.findOne({ documentId, tenantId })
+					.catch((error) => {
+						Lumberjack.error(
+							`Error retrieving checkpoint from local DB.`,
+							lumberProperties,
+							error,
+						);
+					});
+
+				if (checkpoint?.scribe) {
+					lastCheckpoint = JSON.parse(checkpoint.scribe);
+					isLocalCheckpoint = true;
+				} else {
+					lastCheckpoint = JSON.parse(document.scribe);
```

Comment:
Dplicate code.

Reply:
I edited the logic so it is easier to follow and less repetitive